Fix for #243 in ActiveRecord 3.0/3.x

danp60 opened the pull request:
We came across #243 in Rails 3.0 also. This PR applies the fix in #243 and #244 for rails 3.0/3.x and also copies the same spec over.

flyerhzm commented:
@danp60 tests failed on travis-ci, could you fix it?

danp60 commented:
@flyerhzm I just pushed a commit to fix the tests for rails3.0 with ruby2.0. Looks like an incompatibility introduced in ruby 2.0. I used the patch from here: https://github.com/rails/rails/issues/11026#issuecomment-26921562

danp60 commented:
After thinking about this some more, I'm not sure if including that monkey patch is the right way to go, since Rails 3.0 is not supported with Ruby 2.0+. I'm thinking the Rails 3.0 integration should be tested against Ruby 1.9.3. What do you think?

flyerhzm commented:
@danp60 oh, I didn't realize it, let's don't use the patch, just remove rails 3.0 with ruby 2.0+ combination from .travis.yml

danp60 commented:
@flyerhzm I updated closed and re-opened the issue to re-run travis. I also found another bug where `save!` was not patched. I updated this for rails 3, 4, and 5 and added specs for all versions of rails.

danp60 commented:
cc: @tay

flyerhzm commented on Gemfile.rails-3.0:
```diff
@@ -6,6 +6,7 @@ gem 'rails', '~> 3.0.0'
 gem 'sqlite3', platforms: [:ruby]
 gem 'activerecord-jdbcsqlite3-adapter', platforms: [:jruby]
 gem 'activerecord-import'
+gem 'tins', '~> 1.6.0', platforms: [:ruby_19]
```
what's tins for?

danp60 replied:
tins is automatically pulled in by `coveralls`. For ruby_19, I had to lock it to 1.6.0 which is the last version that supports Ruby 1.9

flyerhzm commented on spec/bullet_spec.rb:
```diff
@@ -87,7 +87,7 @@
 
   describe '#add_whitelist' do
-    context 'for ‘special’ class names' do
+    context 'for \'special\' class names' do
```
double quote outside?

danp60 commented:
@flyerhzm I fixed the double quote issue. Thanks!

flyerhzm commented:
@danp60 thanks

connorshea commented:
@flyerhzm @danp60 The use of `alias_method_chain` on [this line](https://github.com/flyerhzm/bullet/pull/278/files#diff-eddadcd35201914b2812f61f0b36a837R39) causes deprecation warnings in Rails 5.0.0 (tested on beta3, should effect all betas).

```
DEPRECATION WARNING: alias_method_chain is deprecated. Please, use Module#prepend instead. From module, you can access the original method using super. (called from block in enable at /Users/connor/.rvm/gems/ruby-2.3.0/gems/bullet-5.0.0/lib/bullet/active_record5.rb:31)
```
